src/main/java/org/hpccsystems/jdbcdriver/HPCCConnection.java

```diff
@@ -584,26 +598,32 @@ public Version getVersion() throws SQLException
         return wsSQLClient.getVersion();
     }
 
-    public ECLWorkunit prepareSQL(String sqlQuery) throws Exception
+    public ECLWorkunitWrapper prepareSQL(String sqlQuery) throws Exception
     {
         if (isClosed())
             throw new SQLException("ERROR: HPCCConnection is closed");
 
         return wsSQLClient.prepareSQL(sqlQuery, targetcluster, queryset, connectTimeoutMillis);
     }
 
-    public ExecutePreparedSQLResponse executePreparedSQL(String wuid, NamedValue[] variables) throws Exception
+    public ExecutePreparedSQLResponseWrapper executePreparedSQL(String wuid, NamedValueWrapper[] variables) throws Exception
     {
         if (isClosed())
             throw new SQLException("ERROR: HPCCConnection is closed");
 
         if (wsSQLClient == null)
             throw new SQLException("ERROR: WsSQLClient not available");
 
-        return wsSQLClient.executePreparedSQL(wuid, targetcluster, variables, readTimoutMillis, eclResultLimit, pageOffset, pageSize, userName, false, false);
+        NamedValue[] namedValueVariables = new NamedValue[variables.length];
```

Comment:
As mentioned above, it's too bad you're forced to do this here, but at least you were able to shield the caller from referencing the NamedValue directly.